source/fundamentals/logging.txt

```diff
@@ -0,0 +1,477 @@
+
+.. code-block:: none
+
+   WARNING: SLF4J not found on the classpath.  Logging is disabled for the 'org.mongodb.driver' component
+
+To set up a logger, you must include the following in your project.
+
+* The ``slf4j-api`` artifact
+* a logging framework 
+* a **binding**
+
+.. note:: 
+
+    For the most popular logging frameworks, there is often a single binding
+    artifact that lists the ``slf4j-api`` and the logging framework as
+    dependencies. This means that you can set up a logger by adding one artifact
+    to your project's dependency list. You will see this in the example below.
+
+A binding is a piece of code that connects the ``slf4j-api`` artifact with a
+logging framework. The example below shows how to bind the ``slf4j-api`` artifact
+to the two most popular logging frameworks, Log4j2 and Logback.
+
+Example - Set Up
+~~~~~~~~~~~~~~~~
+
+This example shows how to set up your logger. Click the
+tab corresponding to the logging framework you would like to use in your project.
+
+.. note:: Dependency Versions
+
+   The versions listed below are illustrative rather than a
+   source of truth. You should check the official documentation for SLF4J and
+   your logging framework of choice for guaranteed up-to-date version
+   information.
+   
+   * `SLF4J documentation <http://www.slf4j.org/>`__
+   * `Logback documentation <http://logback.qos.ch/>`__
+   * `Log4j2 documentation <https://logging.apache.org/log4j/2.x/manual/>`__
+
+.. tabs::
+
+   .. tab:: Logback
+      :tabid: Logback-binding
+
+      Select the build tool you are using in your project.
+
+      .. tabs::
+
+         .. tab:: Maven
+            :tabid: maven
+                        
+            Add the following dependency to your ``pom.xml`` file.
+   
+            .. code-block:: xml
+   
+               <dependency>
+                 <groupId>ch.qos.logback</groupId>
+                 <artifactId>logback-classic</artifactId>
+                 <version>1.2.3</version>
+               </dependency>
+
+         .. tab:: Gradle
+            :tabid: gradle
+   
+            Add the following dependency to your ``build.gradle`` file.
+   
+            .. code-block:: none
+   
+               dependencies {
+                  implementation 'ch.qos.logback:logback-classic:1.2.3'
+               }
+
+      Once you have included the above dependency, connect to your
+      MongoDB instance and retrieve a document with the following code.
+
+      .. include:: /includes/fundamentals/logging.rst
+
+      You should see output that resembles the following:
+
+      .. code-block:: none
+
+         ...
+         1156 [main] DEBUG o.m.driver.protocol.command - Sending command '{"find": "<your collection>", "filter": {}, "limit": 1, "singleBatch": true, "$db": "<your db>", "lsid": {"id": {"$binary": {"base64": "<server session>", "subType": "04"}}}}' with request id 13 to database <your db> on connection <your connection> to server <your server>
```

Comment:
I'm surprised we don't also see INFO level logging in this output.

Reply:
Thanks for the note. I agree that having INFO logging in this output would help illustrate the point for learners. Luckily the third from last from the last line of logging output in the code snippet has a log level of INFO. Adding it now. Fix: https://github.com/mongodb/docs-java/pull/76/commits/6c92b0c431ba28c14dc12aa6b661d65bf2ff9aa9